--- typo3_src-3.8.0/typo3/sysext/indexed_search/modfunc2/class.tx_indexedsearch_modfunc2.php	2005-04-07 10:43:15.000000000 +0200
+++ typo3_src-3.9.0-dev/typo3/sysext/indexed_search/modfunc2/class.tx_indexedsearch_modfunc2.php	2005-06-13 22:33:33.000000000 +0200
@@ -93,6 +93,11 @@
 			.$this->listSeveralStats($LANG->getLL("last24hours"),$addwhere3,$conf).'</td></tr></table>'
 			.$this->note;

+			// Ask hook to include more on the page:
+		if ($hookObj = &$this->hookRequest('additionalSearchStat'))	{
+			$content.= $hookObj->additionalSearchStat();
+		}
+
 		return $content;
 	}

@@ -187,7 +192,25 @@
 		return t3lib_tsfeBeUserAuth::extGetTreeList($id,$depth,$begin,$perms_clause);
 	}

+	/**
+	 * Returns an object reference to the hook object if any
+	 *
+	 * @param	string		Name of the function you want to call / hook key
+	 * @return	object		Hook object, if any. Otherwise null.
+	 * @author Kasper Skaarhoj
+	 */
+	function &hookRequest($functionName)	{
+		global $TYPO3_CONF_VARS;

+			// Hook: menuConfig_preProcessModMenu
+		if ($TYPO3_CONF_VARS['EXTCONF']['indexed_search']['be_hooks'][$functionName]) {
+			$hookObj = &t3lib_div::getUserObj($TYPO3_CONF_VARS['EXTCONF']['indexed_search']['be_hooks'][$functionName]);
+			if (method_exists ($hookObj, $functionName)) {
+				$hookObj->pObj = &$this;
+				return $hookObj;
+			}
+		}
+	}
 }


